paddle/fluid/framework/lod_tensor.h

```diff
@@ -55,7 +55,12 @@ namespace framework {
  *    0 2 4 7
  *    0 2 5 7 10 12 15 20
  */
+#if defined(PADDLE_WITH_CUDA)
```

Comment:
#ifdef PADDLE_WITH_CUDA

Reply:
This only affects NLP models, the NLP tested https://github.com/PaddlePaddle/Paddle/compare/develop...Superjomn:bugfix/lod-Vector2vector, and that saves a lot performance.

Reply:
https://gcc.gnu.org/onlinedocs/cpp/Defined.html

`#if defined` is equivalent to `#ifdef`